### PR TITLE
Add high/medium priority queues and in-app autoscaling

### PR DESCRIPTION
**Pull request by davisagli**
This adds the ability to specify which queue should be used to build a Plan (high, medium, or default). Workers will process all 3 queues, preferring jobs from the `high` queue, then `medium`, then `default`. The METACI_WORKER_RESERVE setting can be used to configure a number of workers that will be reserved to process high priority builds only.

This also adds the ability for the app to handle its own autoscaling instead of relying on an external service. When enabled, the app will check the queues once a minute and add workers when necessary up to the limit specified by the METACI_MAX_WORKERS setting. Enabling autoscaling requires a few configuration steps mentioned in the README.

### CODE REVIEW

File: metaci/build/autoscaling.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+import requests
+from cumulusci.core.utils import import_global
+from django.conf import settings
+from rq import Worker
+from rq.registry import StartedJobRegistry
+
+logger = logging.getLogger(__name__)
+
+
+class Autoscaler(object):
```
Comment (cdcarter) [inline metaci/build/autoscaling.py]: i suspect this base class may want to expose a structured log line for (current_workers, demand, delta) or whatever, something you can extract the metric on heroku of how often you have dynos stilling with no work waiting to scale down

File: README.rst [hunk 2]
```diff
@@ -137,12 +137,18 @@ Notifications
...
+To configure autoscaling:
 
-Configure Hirefire and then run `heroku config:set HIREFIRE_TOKEN=YOUR_TOKEN`
+1. Set the METACI_MAX_WORKERS setting to the maximum number of dynos you'd like to scale up to.
+2. Set the METACI_WORKER_RESERVE setting to the number of dynos you'd like to reserve for high-priority builds. (Optional; defaults to 1.)
+3. Set up a Heroku user with access to this app, and create an authorization token using ``heroku authorizations:create``. Set the HEROKU_TOKEN setting to this authorization token.
+4. Set the HEROKU_APP_NAME setting to the name of the Heroku app.
+5. Add a RepeatableJob with Callable = ``metaci.build.autoscaling.autoscale``, Queue = ``short``, and Interval = ``1 minute``.
```
[the remark this answers is not in the document]
Comment (davisagli) [reply README.rst]: We added a call to autoscale just after a build is queued, so it's not needed for scaling up. But we still need something to notice and scale down when there are no builds left.

File: metaci/build/autoscaling.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+
+class NonAutoscaler(Autoscaler):
+    """Don't actually autoscale."""
+
+
+class LocalAutoscaler(Autoscaler):
+    """Scale by starting rqworker subprocesses in burst mode."""
+
+    processes = []
```
[the remark this answers is not in the document]
Comment (davisagli) [reply metaci/build/autoscaling.py]: I was worried that the process objects would get GC'd and kill the process.

File: metaci/build/autoscaling.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+            LocalAutoscaler.processes = []
+        else:
+            count = self.target_workers - self.count_workers()
+            if count > 0:
+                logger.info(f"Starting {count} workers in burst mode")
+                for x in range(count):
+                    self.processes.append(
+                        subprocess.Popen(
+                            [
+                                "python",
```
[the remark this answers is not in the document]
Comment (davisagli) [reply metaci/build/autoscaling.py]: I mean we ought to be running in a virtualenv or Heroku slug where `python` is what we want.

File: metaci/build/autoscaling.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (cdcarter) [reply metaci/build/autoscaling.py]: is that a thing that python will do? i assumed a popen wouldn't get gc'd til it's closed...but fair worry!

[the remark this answers is not in the document]
Comment (prescod) [reply metaci/build/autoscaling.py]: Python subprocesses can actually outlive their parent processes. But I would generally be inclined towards belt and suspenders in this situation myself...feels weird to create a thing and have no way to refer to it.

**Pull request merged**